struqture/src/bosons/bosonic_indices.rs

```diff
@@ -724,7 +741,24 @@ impl ModeIndex for HermitianBosonProduct {
         creators.sort_unstable();
         let mut annihilators: TinyVec<[usize; 2]> = annihilators.into_iter().collect();
         annihilators.sort_unstable();
-        if creators.iter().min() > annihilators.iter().min() {
+        let mut hermitian_conjugate = false;
+        let mut number_equal_indices = 0;
+        for (creator, annihilator) in creators.iter().zip(annihilators.iter()) {
+            match annihilator.cmp(creator) {
+                std::cmp::Ordering::Less => {
+                    hermitian_conjugate = true;
```

Comment:
Maybe we can also break out of the cycle in this arm

Reply:
Yes, we could not in the validity check which has similar code but here we can